src/vm/gdbjit.cpp

```diff
@@ -1845,12 +1845,29 @@ static inline bool isListedModule(const WCHAR *wszModuleFile)
     LPWSTR pComma = wcsstr(g_wszModuleNames, W(","));
     LPWSTR tmp = g_wszModuleNames;
 
+    // remove '.ni.dll' or '.ni.exe' suffix from wszModuleFile
+    size_t ModuleFileLen = wcslen(wszModuleFile);
+    LPWSTR wszNonNIModuleFile = new WCHAR[ModuleFileLen];
+    LPCWSTR pNIEXE = wcsstr(wszModuleFile, W(".ni.exe")); // where '.ni.exe' start at
+    LPCWSTR pNIDLL = wcsstr(wszModuleFile, W(".ni.dll")); // where '.ni.dll' start at
+    if (pNIEXE) {
+      wcsncpy(wszNonNIModuleFile, wszModuleFile, pNIEXE - wszModuleFile);
+      wszNonNIModuleFile[pNIEXE - wszModuleFile] = W('\0');
+      wcscat(wszNonNIModuleFile, W(".dll"));
```

Comment:
`dotnet build` generates xxx.dll file. Then crossgen generates xxx.ni.exe file for xxx.dll which contains Main entry.  Since gdbjit users usually export CORECLR_GDBJIT=xxx.dll, this extension is ok.